libkatana_python_native/src/Reductions.cpp

```diff
@@ -0,0 +1,95 @@
+#include <katana/Reduction.h>
+#include <katana/python/Conventions.h>
+#include <katana/python/NumbaSupport.h>
+#include <katana/python/PythonModuleInitializers.h>
+#include <katana/python/TemplateSupport.h>
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+
+namespace {
+
+// TODO(amp): Use template template parameters once we can rely on having them.
+struct ForGAccumulator {
+  template <typename T>
+  using type = katana::GAccumulator<T>;
+};
+
+struct ForReduceMin {
+  template <typename T>
+  using type = katana::GReduceMin<T>;
+};
+
+struct ForReduceMax {
+  template <typename T>
+  using type = katana::GReduceMax<T>;
+};
+
+// The following two structs allow GReduceLogicalOr/And to be passed to
+// ReducibleFunctor despite not being templated.
+struct ForReduceLogicalOr {
+  template <typename T>
+  using type = katana::GReduceLogicalOr;
+};
+
+struct ForReduceLogicalAnd {
+  template <typename T>
+  using type = katana::GReduceLogicalAnd;
+};
+
+template <typename For>
+struct ReducibleFunctor {
+  template <typename T>
+  py::object operator()(py::module_& m, const char* name) {
+    py::class_<typename For::template type<T>> cls(
+        m, name,
+        "A reducer object that can updated with new values and combines the "
+        "results efficiently using the appropriate operator.\n"
+        "\n"
+        "This class can be passed into numba compiled code and it's methods "
+        "can be used from there.\n");
+    cls.def(py::init<>());
+    cls.def(
+        py::init([](T v) {
+          auto self = std::make_unique<typename For::template type<T>>();
+          self->update(v);
+          return self;
+        }),
+        "Create a new instance with an initial value.");
+    katana::RegisterNumbaClass(cls);
+    katana::DefWithNumba<const T&>(
+        cls, "update", &For::template type<T>::update,
```

Comment:
So this will call the update method on `GReduceLogicalAnd` or whatever else it's templated with?

Reply:
This line will not call it. But the python method `cls.update` will call `update` on whatever the template argument specifies as you say.